SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Channel/ChannelEventArgs.cs

```diff
@@ -3,33 +3,98 @@
 
 namespace Microsoft.AppCenter.Channel
 {
+    /// <summary>
+    /// Base type for all channel events.
+    /// </summary>
     public abstract class ChannelEventArgs : EventArgs
     {
+        /// <summary>
+        /// Log associated to the event.
+        /// </summary>
+        public Log Log { get; protected set; }
+
+        /// <summary>
+        /// Init event with a log.
+        /// </summary>
+        /// <param name="log">log associated to this event.</param>
         protected ChannelEventArgs(Log log)
         {
             Log = log;
         }
-        public Log Log { get; protected set; }
     }
+
+    /// <summary>
+    /// Event called when a log is enqueued to a channel but before properties are final.
+    /// </summary>
     public class EnqueuingLogEventArgs : ChannelEventArgs
     {
+        /// <summary>
+        /// Init event with a log.
+        /// </summary>
+        /// <param name="log">log associated to this event.</param>
         public EnqueuingLogEventArgs(Log log) : base(log) { }
     }
+
+    /// <summary>
+    /// Event called to possibly filter out a log before it is persisted and scheduled for sending.
+    /// </summary>
+    public class FilteringLogEventArgs : ChannelEventArgs
+    {
+        /// <summary>
+        /// Set this property to true to request the log to be filtered out.
+        /// </summary>
+        public bool FilterRequested { get; set; }
```

Comment:
Have you determined that the order of invocations won't affect whether this property is true or false when channel reads it?

Reply:
Order of invocations do affect the result but this is actually the official c# event pattern with bool return: https://docs.microsoft.com/en-us/dotnet/csharp/event-pattern#returning-values-from-event-subscribers

I even have a unit test to show filters canceling each other, this is by design.